Extending/Property-Editors/Build-a-Block-Editor.md

```diff
@@ -0,0 +1,253 @@
+---
+versionFrom: 8.0.0
+---
+
+# Build a Block Editor
+
+To read this document you must make your self confident with the basics of developing a custom Property Editor for Umbraco. Read more on that in the [Property Editors article](../../Extending/Property-Editors/).
+
+
+## Setup your Property Editor as a Block Property Editor
+
+In order for your editor to become a Block Editor you must setup your property editor through C#.
+
+```csharp
+using ClientDependency.Core;
+using Umbraco.Core.Logging;
+using Umbraco.Core.PropertyEditors;
+using Umbraco.Web.PropertyEditors;
+// TODO: Make sure we dont miss some dependencies?
+
+namespace UmbracoEightExamples.PropertyEditors
+{
+
+    
+    [DataEditor(
+        "MyOwn.UnicornBlocksEditor",
+        "Unicorn Blocks",
+        "unicornblocks",
+        ValueType = ValueTypes.Json,
+        Group = Constants.PropertyEditors.Groups.Lists,
+        Icon = "icon-thumbnail-list")]
+    [PropertyEditorAsset(ClientDependencyType.Javascript, "/App_Plugins/UnicornBlocks/UnicornBlocks.controller.js")]
+    public class UnicornBlocksPropertyEditor : BlockEditorPropertyEditor
+    {
+    	
+    	public UnicornBlocksPropertyEditor(ILogger logger, Lazy<PropertyEditorCollection> propertyEditors, IDataTypeService dataTypeService, IContentTypeService contentTypeService)
+            : base(logger, propertyEditors, dataTypeService, contentTypeService, new DataHelper())
+        { }
+    	
+    }
+
+}
+```
+
+Notice how the `PropertyEditorAsset` attribute is used to load the `UnicornBlocks.controller.js` JavaScript file.
+
+Your Property Editor would as well need a `PropertyValueConverter`.
+
+**TODO: write something about PropertyValueConverter, or provide a link to something about that?**
```

Comment:
FYI, we have pretty good docs on that already here 🙂
https://our.umbraco.com/Documentation/Extending/Property-Editors/value-converters